### DISCUSSION TITLE
Error while editing the existing device.

### DISCUSSION THREAD

**Discussion by Narek307** [netbox-community/netbox#6821] [General]
I am getting the following error: "**Unknown field name 'coordinates' in custom field data.**" while editing the existing device.
I am getting the following error: "Unknown field name 'coordinates' in custom field data." while editing the existing device.
Can you please help me to solve this issue?

**Comment by candlerb** [comment]
What Netbox version are you running? I am presuming it is at least Netbox 2.10.0 which has the new custom field implementation.

It looks to me what's happened is that your `custom_field_data` contains `{"coordinates": "something"}` but you have no custom field called "coordinates" defined on Device.  See [code](https://github.com/netbox-community/netbox/blob/develop/netbox/netbox/models.py#L111-L114).

Workaround 1: create a new custom field called "coordinates" for Device; and then delete it.  That *should* clean up your data.  (However it's unclear how it got into this state in the first place)
Workaround 2: fix up the custom_field_data field in dcim.devices using a Django script or direct SQL update

I think that Netbox ought to clean up data automatically in this case, rather than raising a ValidationError, since it's impossible for the GUI user to fix the data.  You could raise a feature request for this.

**Reply by Narek307** [reply]
Thank you for your answer.
I am running Netbox-2.11.8..
Workaround 1: Where should I create this filed (in which file)?

Thank you for your support in advance.

**Reply by candlerb** [reply]
In the "Admin" section of Netbox, section "Custom Fields".  Create a new custom field called "coordinates", link it to Device.  Then delete it.

**Reply by Narek307** [reply]
It didn't work, thank you for your time.
I will open a feature request.

**Reply by candlerb** [reply]
That's odd.  I can't reproduce the problem with Netbox v2.11.10:

* Add a custom field type Text, name "coordinates", assigned to DCIM > Device
* Edit a device.  Set the coordinates to "foo,bar"
    * `select * from dcim_device where id=<N>` shows that custom_field_data is `{"coordinates": "foo,bar"}`
* Delete the custom field "coordinates"
   * `select * from dcim_device where id=<N>` shows that custom_field_data has been reset to `{}`
* Edit the same device. Change the comment field and save. It's successful.
 
You may find the following SQL query useful to identify the problem:

```
select id,custom_field_data from dcim_device where custom_field_data != '{}';
```

You'll need to get into the PSQL query shell first: e.g. under Ubuntu it's `sudo -u postgres psql netbox`

Can you show an example of some rows where `custom_field_data` contains `"coordinates"` ?

**Reply by Narek307** [reply]
It was related to netbox_topology_views plugin settings. I have reactivated it and then repeated the steps that you have recommended. It is working now. Thank you very much.
